src/date_type.h

```diff
@@ -92,8 +102,11 @@ static const TimerGameCalendar::Year DEF_END_YEAR = ORIGINAL_END_YEAR - 1;
  */
 static const TimerGameCalendar::Year MAX_YEAR  = 5000000;
 
-/** The number of days till the last day */
-#define MAX_DAY (DAYS_TILL(MAX_YEAR + 1) - 1)
+/** Calculate the number of days till the last day, starting at day 0. */
+static inline TimerGameCalendar::Day MaxDay()
```

Comment:
`static const TimerGameCalendar::Day MAX_DAY = DaysTill(MAX_YEAR + 1) - 1;`
may work... may need constexpr. Not sure though.

Reply:
Yes, it works. `constexpr` does not.